RegistryManager/src/main/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceService.java

```diff
@@ -76,453 +63,215 @@ public class CSourceService extends BaseQueryService implements EntryCRUDService
-		@SuppressWarnings("unchecked")
-		String value = (String) ((List<Map<String, Object>>) expiresAt).get(0).get(NGSIConstants.JSON_LD_VALUE);
-		try {
-			return SerializationTools.date2Long(value);
-		} catch (Exception e) {
-			throw new AssertionError("In invalid date time came pass the payload checker");
-		}
-	}
-
-//	public Uni<QueryResult> query(QueryParams qp) {
-//		return csourceInfoDAO.query(qp);
-//	}
-
-	@Override
-	protected StorageDAO getQueryDAO() {
-		return cSourceInfoDAO;
-	}
-
-	@Override
-	protected StorageDAO getCsourceDAO() {
-		// intentional null!!!
-		return null;
-	}
-
-	public void handleEntityDelete(BaseRequest message) {
-		String id = message.getId();
-		String tenant = message.getTenant();
-		Map<String, Object> informationEntry = tenant2EntityId2InformationEntry.remove(tenant, id);
-		Set<String> ids = tenant2InformationEntry2EntityIds.get(tenant, informationEntry);
-		if (ids == null) {
-			return;
-		}
-		ids.remove(id);
-		if (ids.isEmpty()) {
-			tenant2InformationEntry2EntityIds.remove(tenant, informationEntry);
-			try {
-				CSourceRequest regEntry = createInternalRegEntry(tenant);
-				if (regEntry.getFinalPayload() == null) {
-					deleteEntry(regEntry.getHeaders(), regEntry.getId()).await().indefinitely();
-				} else {
-					storeInternalEntry(regEntry);
-				}
-			} catch (Exception e) {
-				logger.error("Failed to store internal registry entry", e);
+	public Uni<NGSILDOperationResult> updateRegistration(String tenant, String registrationId,
+			Map<String, Object> entry) {
+		AppendCSourceRequest request = new AppendCSourceRequest(tenant, registrationId, entry);
+		return cSourceInfoDAO.updateRegistration(request).onItem().transformToUni(rowset -> {
+			if (rowset.rowCount() > 0) {
+				// no need to query regs again they are not distributed
+				// request.setPayload(rowset.iterator().next().getJsonObject(0).getMap());
+				return kafkaSenderInterface.send(request).onItem().transform(v -> {
+					NGSILDOperationResult result = new NGSILDOperationResult(AppConstants.OPERATION_UPDATE_REGISTRATION,
+							registrationId);
+					result.addSuccess(new CRUDSuccess(null, null));
+					return result;
+				});
+			} else {
+				return Uni.createFrom().failure(new ResponseException(ErrorType.NotFound, "Registration not found"));
 			}
-		} else {
-			tenant2InformationEntry2EntityIds.put(tenant, informationEntry, ids);
-		}
-	}
-
-	public void handleEntityCreateOrUpdate(BaseRequest message) {
-		Map<String, Object> informationEntry = getInformationFromEntity(message.getFinalPayload());
-		checkInformationEntry(informationEntry, message.getId(), message.getTenant());
+		}).onFailure().recoverWithUni(
+				// TODO do some proper error handling depending on the sql code
+				e -> {
+					if (e instanceof ResponseException) {
+						return Uni.createFrom().failure((ResponseException) e);
+					} else {
+						return Uni.createFrom().failure(new ResponseException(ErrorType.InternalError, e.getMessage()));
```

Comment:
Proper error handling.

Reply:
send response Exception to caller method.